Objects/typeobject.c

```diff
@@ -4719,6 +4737,15 @@ _PyType_FromMetaclass_impl(
                         goto finally;
                     }
                 }
+                if (strcmp(memb->name, "__weaklistoffset__") == 0) {
+                    weaklistoffset_member = memb;
+                }
+                if (strcmp(memb->name, "__dictoffset__") == 0) {
+                    dictoffset_member = memb;
+                }
+                if (strcmp(memb->name, "__vectorcalloffset__") == 0) {
+                    vectorcalloffset_member = memb;
+                }
```

Comment:
I would prefer to use `if-else` or `continue`.  Non-debug builds on MSVC currently avoid a crash here by relying on a default SSA-optimization option (partial redundancy elimination) for some reason, which is usually disabled (`/d2ssa-pre-`) to diagnose or work around a crash unlike this.

Reply:
Adding `else` sounds like a great change! `continue` would work too.
You could reuse gh-123465 to keep the fix-up commit grouped with this one.